cluster/cluster.go

```diff
@@ -142,7 +145,8 @@ LOOP:
 		}
 		// if no nodes have been selected yet then grab a
 		// random node from the set of available nodes
-		selected := candidates.nodes[rand.Intn(len(candidates.nodes))]
+		pos := int(atomic.AddUint64(&counter, 1)) % len(candidates.nodes)
```

Comment:
should that counter maybe get reset to `0` at some point? I know `Uint64` is big, but being optimistic we should assume MT can have a long life time without restarting :)

Reply:
when it wraps around, all the logic keeps on functioning fine as far as i can see.
in fact i might as well have used a uint32 for that reason.

Reply:
yeah true